src/shared/components/palette/address/address-option.model.ts

```diff
@@ -1,21 +1,32 @@
 import { AddressModel } from '../../../domain/addresses/address.model';
 export class AddressOption {
 
-  value: AddressModel;
   description: string;
 
+  value: AddressModel;
+
   constructor (addressModel: AddressModel, description: string) {
-    this.value = addressModel;
-    this.description = (description == null)
-        ? this.value.AddressLine1
-            +  this.prefixWithCommaIfPresent(this.value.AddressLine2)
-              +  this.prefixWithCommaIfPresent(this.value.AddressLine3)
-                + ', ' + this.value.PostTown
-                  : description;
+    if(description == null) {
+      this.value = addressModel;
```

Comment:
we are setting the addressModel only if `description==null`. Before we were not. Is that what we want?